### PR TITLE
SDIT-1348 Check feature switch status for activities migration preview

### PR DESCRIPTION
**Pull request by mikehalmamoj**
image

### CODE REVIEW

File: server/routes/activitiesMigration/activitiesMigrationController.ts [hunk 1]
```diff
@@ -94,12 +94,21 @@ export default class ActivitiesMigrationController {
...
+        })
+        return 'Error checking ACTIVITY feature switch'
+      }),
+    ]).then(([estimatedCount, dlqCount, incentiveLevels, featureSwitchOn]) => {
       req.session.startActivitiesMigrationForm.estimatedCount = estimatedCount.toLocaleString()
       req.session.startActivitiesMigrationForm.dlqCount = dlqCount.toLocaleString()
       req.session.startActivitiesMigrationForm.incentiveLevelIds = incentiveLevels.map(
         (level: IncentiveLevel) => level.code,
       )
+      req.session.startActivitiesMigrationForm.prisonSwitchedOn = featureSwitchOn
```
Comment (mikehalmamoj) [inline server/routes/activitiesMigration/activitiesMigrationController.ts:111]: This is the main change - check that the prison has been feature switched on during the migration preview.

File: server/routes/activitiesMigration/activitiesMigrationController.ts [hunk 2]
```diff
@@ -124,6 +133,12 @@ export default class ActivitiesMigrationController {
     res.redirect('/activities-migration')
   }
 
+  async postActivatePrison(req: Request, res: Response): Promise<void> {
+    const { serviceName, prisonId } = req.query as { serviceName: string; prisonId: string }
+    await this.nomisPrisonerService.createServiceAgencySwitch(prisonId, serviceName, context(res))
+    res.redirect('/activities-migration/amend')
+  }
+
```
Comment (mikehalmamoj) [inline server/routes/activitiesMigration/activitiesMigrationController.ts:141]: If the prison is not turned on we show a button that calls NOMIS API to switch on the prison.

File: integration_tests/e2e/startActivitiesMigration.cy.ts [hunk 3]
```diff
@@ -113,5 +107,83 @@ context('Start Activities Migration', () => {
 
       Page.verifyOnPage(StartActivitiesMigrationConfirmationPage)
     })
+
+    it('Shows successful preview check details', () => {
```
Comment (mikehalmamoj) [inline integration_tests/e2e/startActivitiesMigration.cy.ts:111]: Most of the PR is adding the integration tests for the preview checks.

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 4]
```diff
@@ -425,7 +425,7 @@ const defaultSentencingFailures = {
   ],
 }
 
-const stubHealth = (): SuperAgentRequest =>
+const stubHealth = (failures: string = '153'): SuperAgentRequest =>
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:428]: trivial - could be a number?

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 5]
```diff
@@ -447,7 +447,7 @@ const stubHealth = (): SuperAgentRequest =>
               messagesInFlight: '0',
               dlqStatus: 'UP',
               dlqName: 'dps-syscon-dev-visitsmigration_dlq',
-              messagesOnDlq: '153',
+              messagesOnDlq: `${failures}`,
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:450]: surprised that intellij allowed this - shouldn't it just be
```suggestion
              messagesOnDlq: failures,
```

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 6]
```diff
@@ -458,7 +458,7 @@ const stubHealth = (): SuperAgentRequest =>
               messagesInFlight: '0',
               dlqStatus: 'UP',
               dlqName: 'dps-syscon-dev-sentencingmigration_dlq',
-              messagesOnDlq: '153',
+              messagesOnDlq: `${failures}`,
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:461]:
```suggestion
              messagesOnDlq: failures,
```

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 7]
```diff
@@ -469,7 +469,7 @@ const stubHealth = (): SuperAgentRequest =>
               messagesInFlight: '0',
               dlqStatus: 'UP',
               dlqName: 'dps-syscon-dev-appointmentsmigration_dlq',
-              messagesOnDlq: '153',
+              messagesOnDlq: `${failures}`,
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:472]:
```suggestion
              messagesOnDlq: failures,
```

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 8]
```diff
@@ -480,7 +480,7 @@ const stubHealth = (): SuperAgentRequest =>
               messagesInFlight: '0',
               dlqStatus: 'UP',
               dlqName: 'dps-syscon-dev-activitiesmigration_dlq',
-              messagesOnDlq: '153',
+              messagesOnDlq: `${failures}`,
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:483]:
```suggestion
              messagesOnDlq: failures,
```

File: integration_tests/mockApis/nomisMigrationApi.ts [hunk 9]
```diff
@@ -491,7 +491,7 @@ const stubHealth = (): SuperAgentRequest =>
               messagesInFlight: '0',
               dlqStatus: 'UP',
               dlqName: 'dps-syscon-dev-allocationsmigration_dlq',
-              messagesOnDlq: '153',
+              messagesOnDlq: `${failures}`,
```
Comment (petergphillips) [inline integration_tests/mockApis/nomisMigrationApi.ts:494]:
```suggestion
              messagesOnDlq: failures,
```

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**